src/utils/imageUtils.ts

```diff
@@ -2,6 +2,23 @@ import get from 'lodash/get';
 import Assets from '../assets';
 import type {ImageSourceType} from '../components/image';
 
+export function extractImageSource(source?: ImageSourceType) {
+  const sourceUri = get(source, 'uri');
+
+  if (sourceUri) {
+    // Case when uri is a number (Local Assets)
+    if (typeof sourceUri === 'number') {
```

Comment:
I personally don't like this solution, it changes the behavior of a local asset. You are changing the way local assets are passed which is not a good idea (for instance, you can see that ColorSwatch is broken now because it's not using our Icon component)